src/core/connection.c

```diff
@@ -3840,7 +4082,20 @@ QuicConnRecvFrames(
             } else if (Status == QUIC_STATUS_OUT_OF_MEMORY) {
                 return FALSE;
             } else {
-                if (Status != QUIC_STATUS_INVALID_STATE) {
+                if (Status == QUIC_STATUS_VER_NEG_ERROR) {
+                    if (QuicBindingQueueStatelessOperation(
+                            Connection->Paths[0].Binding,
+                            QUIC_OPER_TYPE_VERSION_NEGOTIATION,
+                            CxPlatDataPathRecvPacketToRecvData(Packet))) {
+                        //
+                        // Packets that elicit a VN response don't get counted as received
+                        // if they're deferred. Update that count here to ensure it gets counted.
+                        //
+                        Connection->Stats.Recv.TotalPackets += 2;
+                        Packet->ReleaseDeferred = TRUE;
+                    }
+                    QuicConnTransportError(Connection, QUIC_ERROR_VERSION_NEGOTIATION_ERROR);
```

Comment:
Does this send a connection close in response? You need to make sure it doesn't.

Reply:
It doesn't appear to. But it eventually calls into `QuicStreamRecvShutdown` which queues up a stop sending frame. Since the server is sending a version negotiation packet (to do incompatible version negotiation), it should close the connection silently and immediately, correct?